src/main/java/org/traccar/protocol/Minifinder2ProtocolDecoder.java

```diff
@@ -180,6 +180,9 @@ protected Object decode(
                         if (length == 5) {
                             position.setDeviceTime(new Date(buf.readUnsignedIntLE() * 1000));
                         }
+                        if (alarm > 0) {
+                            position.set("alarmCode", alarm);
```

Comment:
Why not decode it properly? Are you missing some specific alarms?

Reply:
There are 18 alarms available for Minifinder2 devices, but most of them are quite specific. 
I could decode them properly but this would add a lot of `ALARM_...` fields in Position.java. 
Or I could hardcode the alarm names in the Minifinder2 decoder. What would you prefer?

Reply:
What alarms are we talking about? Please provide the protocol documentation.

Reply:
<img width="451" alt="Scherm­afbeelding 2024-11-09 om 14 50 14" src="https://github.com/user-attachments/assets/5630dd27-2218-4dae-af01-6005bdb939ae">

Reply:
A lot of these can be mapped to the standard alerts. And some of them don't seem like they should be alerts. For example, we can probably save things like "motion" and "no motion" as the standard motion attribute.

Reply:
I agree, but I also think it would be nice to have the raw alarmCode for the edge cases. The same way this is done with the status information in line 260:
https://github.com/traccar/traccar/blob/2c14a8ebd162cfa21d4e366a1d3d988c0423b320/src/main/java/org/traccar/protocol/Minifinder2ProtocolDecoder.java#L248-L261



I'm more than happy to implement the alerts that can be mapped to standard alerts. How about:

| Bit No. | Description                       | Already implemented as | To implement as                                |
|---------|-----------------------------------|------------------------|------------------------------------------------|
| 0       | Low Battery                       | `ALARM_LOW_BATTERY`    |                                                |
| 1       | Overspeed Alert                   | `ALARM_OVERSPEED`      |                                                |
| 2       | Fall Down Alert                   | `ALARM_FALL_DOWN`      |                                                |
| 3       | Tilt Alert                        |                        | _Maybe add new alarm type?_                      |
| 4       | GEO-1 Alert                       |                        | `ALARM_GEOFENCE_ENTER` `ALARM_GEOFENCE_EXIT` * |
| 5       | GEO-2 Alert                       |                        | `ALARM_GEOFENCE_ENTER` `ALARM_GEOFENCE_EXIT` * |
| 6       | GEO-3 Alert                       |                        | `ALARM_GEOFENCE_ENTER` `ALARM_GEOFENCE_EXIT` * |
| 7       | GEO-4 Alert                       |                        | `ALARM_GEOFENCE_ENTER` `ALARM_GEOFENCE_EXIT` * |
| 8       | Power Off Alert                   | `ALARM_POWER_OFF`      |                                                |
| 9       | Power On Alert                    | `ALARM_POWER_ON`       |                                                |
| 10      | Motion Alert                      |                        | `ALARM_MOVEMENT`                               |
| 11      | No-Motion Alert                   |                        | _Maybe add new alarm type?_                      |
| 12      | SOS-Key                           | `ALARM_SOS`            |                                                |
| 13      | Side1 Key                         |                        |                                                |
| 14      | Side2 Key                         |                        |                                                |
| 15      | Battery charging “Start charging” |                        | `Position.KEY_CHARGE` true                     |
| 16      | Battery no charging               |                        | `Position.KEY_CHARGE` false                    |
| 17      | SOS Completed                     |                        |                                                |
| 18      | Reserved                          |                        |                                                |
| 19      | Reserved                          |                        |                                                |
| 20      | Reserved                          |                        |                                                |
| 21      | Fall down Alert Completed         |                        |                                                |
| 22      | Reserved                          |                        |                                                |
| 23      | Reserved                          |                        |                                                |
| 24      | Reserved                          |                        |                                                |
| 25      | Reserved                          |                        |                                                |

*) using bits 26-29 to determine the direction of the geo alerts.

Reply:
- Motion Alert should map to "motion" = true (can also add as an alarm)
- No-Motion Alert should map to "motion" = false
- Side1/2 Key - can probably add a custom attribute for this
- Tilt Alert - we can probably add a new alarm; I think I've seen it in other devices as well

Reply:
Not sure how to add the Tilt alarm, is it sufficient to add a field in Position.java?

Reply:
No, you have to add it to the template file as well and localization in the web app repo.